testing/scenario_app/ios/Scenarios/ScenariosTests/FlutterEngineTest.m

```diff
@@ -0,0 +1,31 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/Flutter.h>
+#import <XCTest/XCTest.h>
+#import "AppDelegate.h"
+
+@interface FlutterEngineTest : XCTestCase
+@end
+
+@implementation FlutterEngineTest
+
+- (void)testIsolateId {
+  FlutterEngine* engine = [[FlutterEngine alloc] initWithName:@"test" project:nil];
+  XCTAssertNil(engine.isolateId);
+  [self keyValueObservingExpectationForObject:engine keyPath:@"isolateId" handler:nil];
+
+  XCTAssertTrue([engine runWithEntrypoint:nil]);
+
+  [self waitForExpectationsWithTimeout:30.0 handler:nil];
```

Comment:
nit: 30s is too long for a light-weight unit test, and slows down CI progress when there is a true timeout. IMO internally our TAP presubmit should not take more than 1s for loading a Flutter application, so 5s should be enough?

Reply:
This test runs on a physical device or a simulator. It should never take 30s, but I'd rather it not ever flake due to this - 30s should be enough without ever being not enough.

Reply:
FWIW, we've had tests run on hardware that's just bogged down because the machine is under heavy load and it takes a while.

Reply:
OK I see, thanks for explanation.